src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,145 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { useRef } from 'react';
+import { NumberInputIncrement, NumberInputProps } from './types';
+import { merge } from '@utilities/merge';
+import { FOCUS_STYLE } from '@utilities/focusStyle';
+import { useFocusRing } from '@react-aria/focus';
```

Comment:
I would prefer to not use `react-aria` if possible. We are trying to get rid of it and will never achieve it if we keep using it.

Reply:
Is there another preferred approach to focusing that will be replacing react-aria?

Reply:
Try the same way you used `focus-within:` in the same div. 
In `focusStyle.ts` You'll find this :
```
export const FOCUS_VISIBLE_STYLE =
    'focus-visible:tw-ring-4 focus-visible:tw-ring-blue focus-visible:tw-ring-offset-2 focus-visible:dark:tw-ring-offset-black focus-visible:tw-outline-none';
    ```
    
   Try the same but with `focus-within` and remove react-aria.